### ISSUE TITLE
CI: pasta: Couldn't drop cap 3

### ISSUE THREAD

**Issue by edsantiago** [containers/podman#23478]
```
[+0500s] not ok 220 [070] podman build - workdir, cmd, env, label in 4831ms
...
<+221ms> # $ podman run --rm build_test stat -c%u:%g:%N /a/b/c/badsymlink /a/b/c/goodsymlink /bin/mydefaultcmd /a/b/c/myfile
<+180ms> # Error: pasta failed with exit code 1:
         # Couldn't drop cap 3 from bounding set
         # : No child processes
<+005ms> # [ rc=126 (** EXPECTED 0 **) ]
```
In [rawhide rootless](https://api.cirrus-ci.com/v1/artifact/task/4542595240558592/html/sys-podman-rawhide-rootless-host-sqlite.log.html#t--00220). In my parallel-systest PR, so this may be parallel-related or maybe not. Until we have reason to believe otherwise, I'm treating is as a parallel-CI issue.

**Comment by sbrivio-rh** [comment]
Weird: we fail to drop CAP_FOWNER (why are we even starting with it?) and prctl() sets errno to 10, ECHILD (which is not defined as an error for prctl()).

@edsantiago, did this "Couldn't drop cap ..." happen more than once, perhaps with different error numbers and error strings?

**Comment by Luap99** [comment]
> we fail to drop CAP_FOWNER (why are we even starting with it?) 

Because rootless podman runs in a user namespace with all capabilities set so we fork/exec pasta has all caps as well.
```
$ podman unshare grep Cap /proc/self/status 
CapInh: 0000000000000000
CapPrm: 000001ffffffffff
CapEff: 000001ffffffffff
CapBnd: 000001ffffffffff
CapAmb: 0000000000000000
```

But ECHILD does not sounds like a logical error at all, did this fail more than once?

**Comment by edsantiago** [comment]
The string "drop cap " (note the space) only appears once in my flake logs. I'll follow up if I see it again.

**Comment by sbrivio-rh** [comment]
> But ECHILD does not sounds like a logical error at all

Uh-oh, we have a signal handler, pasta_child_handler(), calling functions that might set errno. If that runs while we're dropping capabilities, because the command in Podman runs fast enough (and it doesn't need connectivity), we'll botch errno there.

I'll post a patch to fix that.

**Comment by sbrivio-rh** [comment]
This is now fixed in Debian unstable ([0.0~git20240806.ee36266-1](https://tracker.debian.org/news/1552475/accepted-passt-00git20240806ee36266-1-source-into-unstable/)), Fedora 40 ([passt-0^20240806.gee36266-1.fc40](https://bodhi.fedoraproject.org/updates/FEDORA-2024-d127500a41)), and Fedora Rawhide ([passt-0^20240806.gee36266-1.fc41](https://bodhi.fedoraproject.org/updates/FEDORA-2024-9c77aaa0a3)).
